ast/term_bench_test.go

```diff
@@ -50,6 +50,28 @@ func BenchmarkSetIntersection(b *testing.B) {
 	}
 }
 
+func BenchmarkSetIntersectionDifferentSize(b *testing.B) {
+	sizes := []int{4, 50, 500, 5000}
+	for _, n := range sizes {
+		b.Run(fmt.Sprint(n), func(b *testing.B) {
+			setA := NewSet()
+			setB := NewSet()
+			for i := 0; i < n; i++ {
```

Comment:
To fix the test failure you could do something like this:

```
for i := 0; i < n; i++ {
  setA.Add(IntNumberTerm(i))
}
for i := 0; i < sizes[0]; i++ {
  setB.Add(IntNumberTerm(i))
}
setB.Add(IntNumberTerm(-1))
```

Then in the inner loop assert that len(setC) equals sizes[0].

Reply:
I don't know why I didn't see that. It should have been fixed now